File: 2_ad-hoc/2421/2421.cpp
/*
 * BEE 2421 - Álbum de Fotos - Nível 8 - Ad-hoc
 */
#include <iostream>
using namespace std;

int main()
{
    int x, y, l0, h0, l1, h1;

    cin >> x >> y >> l0 >> h0 >> l1 >> h1;

    if (
        l0 + l1 <= x and max(h0, h1) <= y or
        l0 + h1 <= x and max(h0, l1) <= y or
        l0 + l1 <= y and max(h0, h1) <= x or
        l0 + h1 <= y and max(h0, l1) <= x or
        max(l0, l1) <= x and h0 + h1 <= y or
        max(l0, h1) <= x and h0 + l1 <= y or
        max(l0, l1) <= y and h0 + h1 <= x or
        max(l0, h1) <= y and h0 + l1 <= x)

        cout << 'S' << endl;
    else
        cout << 'N' << endl;
}

File: ad-hoc/2421/2421.py
"""
BEE 2421 - Álbum de Fotos - Level 8 - Ad-hoc
"""

x,y = [int(i) for i in input().strip().split(" ")]
l0, h0 = [int(i) for i in input().strip().split(" ")]
l1, h1 = [int(i) for i in input().strip().split(" ")]

condiction = (
    l0 + l1 <= x and max(h0, h1) <= y or
    l0 + h1 <= x and max(h0, l1) <= y or
    l0 + l1 <= y and max(h0, h1) <= x or
    l0 + h1 <= y and max(h0, l1) <= x or
    max(l0, l1) <= x and h0+h1 <= y or
    max(l0, h1) <= x and h0+l1 <= y or
    max(l0, l1) <= y and h0+h1 <= x or
    max(l0, h1) <= y and h0+l1 <= x
    )

if condiction:
    print("S")
else:
    print("N")
